keps/sig-architecture/3138-increase-reliability-bar/README.md

```diff
@@ -0,0 +1,415 @@
+clusters before.
+
+To make Kubernetes (or any other project) sustainable, we should ensure that
+existing architecture, individual features, tests etc. really works as expected.
+If there is work needed to improve reliability (or stability) of those, it
+should come before new features.
+This is often not the case currently. We significantly rely on the heroic
+efforts of very few individuals that are keeping us on the surface with new bugs,
+issues, test flakes etc. This approach isn’t sustainable and doesn’t scale.
+We need to make it a business of everyone in the whole community.
+
+[Production Readines Survey]: https://datastudio.google.com/reporting/2e9c7439-202b-48a9-8c57-4459e0d69c8d/page/GC5HB
+[How to Break your Cluster with networking]: https://www.youtube.com/watch?v=7qUDyQQ52Uk)
+[1]: https://www.youtube.com/watch?v=QKI-JRs2RIE
+[2]: https://www.youtube.com/watch?v=6sDTB4eV4F8
+[this example]: https://github.com/kubernetes/kubernetes/pull/98305
+
+
+### Goals
+
+- Define the high-level areas of investments
+- Define the sustainable process to ensure reliability investments
+- Define the details of the first milestone as a pilot (reducing test flakiness)
+
+
+### Non-Goals
+
+- Providing exact list of long-standing reliability issues (milestone 3) or new
+  reliability investments (milestone 4) - those will be handled by
+  Reliability WG separately
+
+
+## Proposal
+
+This document outlines steps to improve the quality and reliability of Kubernetes.
+Given the maturity level the project is reaching and the non-negligible amount of
+technical debt accumulated over the years, the most important aspect of the
+proposal is to build the culture which prioritizes certain types of contributions
+over purely feature work.
+
+We have to acknowledge that fixing, and especially debugging, reliability or
+stability issues is the kind of work that very few people like doing - for the
+majority it’s something that they would never do if they wouldn’t have to.
+This is especially true when it comes to debugging flaky tests.
+Moreover, while in companies there are multiple ways of enforcing that certain
+types of work is actually happening (e.g. oncall rotations, dedicated sprints, etc.),
+it’s much harder to achieve in the projects driven by the open-source community.
+
+As a result, we’re proposing introducing a process that will allow us to ensure
+that this kind of work that generally attracts fewer developers  will be really
+happening. This means, that for every piece of such work, we will introduce a two
+phase approach:
+1. announce the reliability-related issue to the SIG asking for prioritizing it
+   with a realistic (and non-conservative) deadline.
+   *We really hope that this will be effective in the majority of cases and
+   we will never be forced to use the point (2) special powers.*
+1. if the issue (reported above) is not addressed before the deadline, block
+   new features from the owning SIG until the issue will get addressed
```

Comment:
The thing that concerns me about blocking all new features is that it's a bit of a chicken and egg situation. SIGs are generally understaffed, they need new contributors to help. New contributors generally get allocated time if they can work on something substantive like a feature. Blocking all new features to a SIG can turn off their new contributor pipeline that they need to tackle all of these reliability issues.

I would suggest an alternative might be: _limiting_ the number of new features based on historical numbers and maybe adding a % carve out for new contributor features. Not wedded to this particular idea, but I think being a little more creative and not turning off the spigot of new contributors entirely should be a partial goal here.